Assets/Scripts/BossStageAlter.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BossStageAlter : MonoBehaviour
+{
+    AudioSource AS;
+    [SerializeField] private GameObject warningPanel;
+    [SerializeField] private AudioClip bgm;
+    [SerializeField] public static bool isWarningOn;
+
+    public void Awake()
+    {
+        isWarningOn = true;
+        //pause the game
+        Time.timeScale = 0;
+        AS = GetComponentInParent<AudioSource>();
+        AS.playOnAwake = true;
+    }
+
+    public void Update()
+    {
+        if (Input.GetKeyDown(KeyCode.Return))
+        {
+            AS.clip = bgm;
+            AS.Play();
+            warningPanel.SetActive(false);
+            Time.timeScale = 1;
+            gameObject.GetComponent<BossStageAlter>().enabled = false;

```

Comment:
it's not a good idea to do GetComponent in update if you can avoid it! it's an expensive method call. You can create a variable for the component and assign it in start, then reference your variable to enable/disable the component